src/components/DatePickerRange.react.js

```diff
@@ -0,0 +1,306 @@
+ * creating datetime objects.
+ *
+ * This component is based off of Airbnb's react-dates react component
+ * which can be found here: https://github.com/airbnb/react-dates
+ */
+export default class DatePickerRange extends Component {
+    constructor(props) {
+      super(props);
+      this.props.startDate = this.props.start_date;
+      this.props.endDate = this.props.end_date;
+      this.props.initialVisibleMonth = this.props.initial_visible_month;
+      this.props.minDateRange = this.props.min_date_range;
+      this.props.maxDateRange = this.props.max_date_range;
+      const momentProps = this.convertPropsToMoment(props);
+      this.state = {
+        focusedInput: props.autoFocusEndDate ? momentProps.startDate : momentProps.endDate,
+        startDate: momentProps.startDate,
+        endDate: momentProps.endDate,
+        initialVisibleMonth: momentProps.initialVisibleMonth,
+        minDateRange: momentProps.min,
+        maxDateRange: momentProps.max,
+        initialStartDate: this.props.start_date,
+        initialEndDate: this.props.end_date,
+        firstInitialVisibleMonth: this.props.initial_visible_month,
+        initialMinDateRange: this.props.min_date_range,
+        initialMaxDateRange: this.props.max_date_range
+      };
+    }
+
+    convertPropsToMoment(props) {
+      let startDate, endDate = null;
+      let initialVisibleMonth = moment(props.startDate);
+      if (typeof props.startDate !== 'undefined') {
+        startDate = moment(props.startDate);
+      }
+      if (typeof props.endDate !== 'undefined') {
+        endDate = moment(props.endDate);
+      }
+      if (typeof props.initialVisibleMonth !== 'undefined') {
+        initialVisibleMonth = moment(props.initialVisibleMonth);
+      }
+      let min, max;
+      if (typeof props.minDateRange !== 'undefined' &&
+          typeof props.maxDateRange !== 'undefined') {
+        min = moment(props.minDateRange);
+        max = moment(props.maxDateRange);
+      }
+      try {
+        if (startDate.isAfter(endDate)) {
+          endDate = null;
+        }
+      } catch (TypeError) {
+        // Continue regardless of error
+      }
+
+      return {startDate, endDate, initialVisibleMonth, min, max}
+    }
+
+    componentWillReceiveProps(newProps) {
+      let propObj = {
+        startDate: newProps.start_date,
+        endDate: newProps.end_date,
+        initialVisibleMonth: newProps.initial_visible_month,
+        minDateRange: newProps.min_date_range,
+        maxDateRange: newProps.max_date_range
+      }
+      const momentProps = this.convertPropsToMoment(propObj);
+      if (this.state.initialStartDate != newProps.start_date) {
+        this.setState({
+          initialStartDate: newProps.start_date,
+          startDate: momentProps.startDate
+        });
+      }
+      if (this.state.initialEndDate != newProps.end_date) {
+        this.setState({
+          initialEndDate: newProps.end_date,
```

Comment:
should this name be changed? `initial` to me means that it's only set once, when the component is rendered. But in this case, we're re-setting the value everytime the component gets new props.